packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/StackScripts/UserDefinedFields/utilities.ts

```diff
@@ -0,0 +1,73 @@
+import type { UserDefinedField } from '@linode/api-v4';
+
+/**
+ * Used to separate required UDFs from non-required ones
+ *
+ * @returns nested array [[...requiredUDFs], [...nonRequiredUDFs]]
+ */
+export const separateUDFsByRequiredStatus = (udfs: UserDefinedField[] = []) => {
+  return udfs.reduce(
+    (accum, udf) => {
+      if (getIsUDFOptional(udf)) {
+        return [[...accum[0]], [...accum[1], udf]];
+      } else {
+        return [[...accum[0], udf], [...accum[1]]];
+      }
```

Comment:
Nit: no need to spread the array that isn't being changed
```suggestion
      if (getIsUDFOptional(udf)) {
        return [accum[0], [...accum[1], udf]];
      } else {
        return [[...accum[0], udf], accum[1]];
      }
```

Reply:
Good catch. For the record, I didn't write that function, it was copy paste from existing StackScript code 😅